src/coap_client.c

```diff
@@ -162,9 +177,24 @@ static enum golioth_status golioth_coap_client_set_internal(struct golioth_clien
 
     if (is_synchronous)
     {
-        // Created here, deleted by coap thread (or here if fail to enqueue
+        // Created here, deleted by coap thread (or here if fail to create or enqueue)
         request_msg.request_complete_event = golioth_event_group_create();
+        if (!request_msg.request_complete_event)
+        {
+            GLTH_LOGW(TAG, "Failed to create event group");
+            request_msg.status = NULL;
```

Comment:
Same comment as above, this isn't necessary (throughout this, the same is true).

Reply:
Great, I've made a subsequent commit to remove similar code not previously touched by this PR.